kse/src/org/kse/gui/dialogs/DViewDHParameters.java

```diff
@@ -0,0 +1,210 @@
+import java.text.MessageFormat;
+import java.util.ResourceBundle;
+
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.JTextArea;
+import javax.swing.SwingUtilities;
+import javax.swing.filechooser.FileNameExtensionFilter;
+
+import org.bouncycastle.util.io.pem.PemObject;
+import org.bouncycastle.util.io.pem.PemWriter;
+import org.kse.ApplicationSettings;
+import org.kse.crypto.CryptoException;
+import org.kse.gui.CurrentDirectory;
+import org.kse.gui.CursorUtil;
+import org.kse.gui.JEscDialog;
+import org.kse.gui.JavaFXFileChooser;
+import org.kse.gui.LnfUtil;
+import org.kse.gui.PlatformUtil;
+import org.kse.utilities.DialogViewer;
+import org.kse.utilities.pem.PemInfo;
+import org.kse.utilities.pem.PemUtil;
+
+import net.miginfocom.swing.MigLayout;
+
+public class DViewDHParameters extends JEscDialog {
+	
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -1711814777923997727L;
+
+	private static ResourceBundle res = ResourceBundle.getBundle("org/kse/gui/dialogs/resources");
+	
+	private JTextArea jtAreaPem;
+	private JButton jbExport;
+	private JButton jbOK;
+	private JButton jbCopy;
+	private static int keySize;
+	private byte [] dhParameters;
+	private static final String FILE_SUFFIX = ".pem";
+	private static final String EB = "DH PARAMETERS"; //Encapsulation Boundary
+	
+	
+	public DViewDHParameters (JFrame parent, String title, byte[] dhParams)
+		throws CryptoException {
+
+		super(parent, title, Dialog.ModalityType.DOCUMENT_MODAL);
+		this.dhParameters = dhParams;
+		initComponents(dhParameters);
+	}
+	
+	private void initComponents(byte[] dhParams) throws CryptoException {
+		//TODO Generate DH Parameters icon
+	    PemInfo pemInfo = new PemInfo(EB, null, dhParams);
+		jtAreaPem = new JTextArea(PemUtil.encode(pemInfo));
+		jtAreaPem.setToolTipText(res.getString("DViewDHParameters.jtAreaPem.tooltip"));
+		jtAreaPem.setEditable(false);
+		jtAreaPem.setFont(new Font(Font.MONOSPACED, Font.PLAIN, LnfUtil.getDefaultFontSize()));
+		// keep uneditable color same as editable
+		jtAreaPem.putClientProperty("JTextArea.infoBackground", Boolean.TRUE);	
+
+		jbExport = new JButton(res.getString("DViewDHParameters.jbExport.export.text"));
+		jbExport.setToolTipText(res.getString("DViewDHParameters.jbExport.export.tooltip"));
+		PlatformUtil.setMnemonic(jbExport, res.getString("DViewDHParameters.jbExport.mnemonic").charAt(0));
+		
+		jbCopy = new JButton(res.getString("DViewDHParameters.jbCopy.text"));
+		jbCopy.setToolTipText(res.getString("DViewDHParameters.jbCopy.tooltip"));
+		PlatformUtil.setMnemonic(jbCopy, res.getString("DViewDHParameters.jbCopy.mnemonic").charAt(0));
+
+		jbOK = new JButton(res.getString("DViewDHParameters.jbOK.text"));
+		
+		Container pane = getContentPane();
+		pane.setLayout(new MigLayout("insets 10"));
+		pane.add(jtAreaPem, "span");
+		pane.add(jbCopy);
+		pane.add(jbExport);
+		pane.add(jbOK);
```

Comment:
It's better to tag the buttons, so Miglayout can decide where to put them to match the platform (order is different on Windows, macOS and Linux):
```
		pane.add(jtAreaPem, "span, wrap");
		pane.add(jbCopy, "tag other");
		pane.add(jbExport, "tag other");
		pane.add(jbOK, "tag ok");
```

Reply:
Updated, good copy on the tag that was useful.